### PR TITLE
Ram/Interpreter Refactoring  (see Issue #541)

### PR DESCRIPTION
**Pull request by b-scholz**
Move the issuing of the RAM debugging report to AstTranslator.cpp

Further refactoring of RAM required to rewrite the interface of Interpreter before continuing the refactoring of RAM.

Disabled auto-scheduling option for the time being. The auto-scheduling framework needs to be rewritten.

### CODE REVIEW

File: src/Interpreter.h [hunk 1]
```diff
@@ -16,743 +16,128 @@
...
     bool hasRelation(const std::string& name) const {
         return data.find(name) != data.end();
     }
 
-    /**
-     * Deletes the referenced relation from this environment.
-     */
+    /** Drop relation */
     void dropRelation(const RamRelation& id) {
         data.erase(id.getName());
```
Comment (mmcgr) [inline src/Interpreter.h]: The relation should be deleted before the entry is erased. We're currently leaking memory.

Comment (b-scholz) [reply src/Interpreter.h]: Good point...

Comment (mmcgr) [reply src/Interpreter.h]: It seems that we also sometimes drop relations that don't exist. I'm not sure if this is intended, and dropRelation should handle it, or unintended and there is a bug elsewhere.

Comment (mmcgr) [reply src/Interpreter.h]: Oh, but it looks like you may have already fixed that.

File: src/Makefile.am [hunk 2]
```diff
@@ -123,6 +121,8 @@ souffle_sources = \
               WriteStreamCSV.h                          \
               IOSystem.h                                \
               Interpreter.cpp       Interpreter.h       \
+			  InterpreterContext.h                      \
```
Comment (mmcgr) [inline src/Makefile.am]: Use spaces instead of tabs here

File: src/InterpreterRelation.h [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+    size_t arity;
+
+    /** Size of blocks containing tuples */
+    static const int BLOCK_SIZE = 1024;
+
+    /** Block data structure for storing tuples */
+    struct Block {
+        size_t size;
+        size_t used;
+        // TODO (#421): replace linked list by STL linked list
```
Comment (mmcgr) [inline src/InterpreterRelation.h]: Is this really #421?

File: src/InterpreterRelation.h [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+        indices.swap(other.indices);
+
+        allocatedBlocks.swap(other.allocatedBlocks);
+    }
+
+    virtual ~InterpreterRelation() {
+        for (auto x : allocatedBlocks) delete[] x;
+    }
+
+    // TODO (#421): check whether still required
```
Comment (mmcgr) [inline src/InterpreterRelation.h]: Is this really #421?

File: src/Makefile.am (hunk 2 above)
Comment (b-scholz) [reply src/Makefile.am]: Done.

File: src/InterpreterRelation.h (hunk 3 above)
Comment (b-scholz) [reply src/InterpreterRelation.h]: Renamed to 541.

File: src/InterpreterRelation.h (hunk 4 above)
Comment (b-scholz) [reply src/InterpreterRelation.h]: Renamed to 541

**Pull request merged**